packages/sw-cli/src/cli/index.js

```diff
@@ -370,14 +388,164 @@ class SWCli {
+      throw err;
+    });
+  }
+
+  _buildFileManifestFromGlobs(manifestFilePath, globs) {
+    let globbedFiles = [];
+    globs.forEach((globPattern) => {
+      const files = this._getFileManifestDetails(globPattern);
+      globbedFiles = globbedFiles.concat(files);
+    });
+
+    const manifestEntries = this._filterFiles(globbedFiles);
+
+    return this._writeFilemanifest(manifestFilePath, manifestEntries);
+  }
+
+  _writeFilemanifest(manifestFilePath, manifestEntries) {
+    try {
+      mkdirp.sync(path.dirname(manifestFilePath));
+    } catch (err) {
+      logHelper.error(errors['unable-to-make-manifest-directory'], err);
+      return Promise.reject(err);
+    }
+
+    return new Promise((resolve, reject) => {
+      fs.readFile(
+        path.join(__dirname, '../lib/templates/file-manifest.js.tmpl'), 'utf8',
+        (err, data) => {
+        if (err) {
+          logHelper.error(errors['read-manifest-template-failure'], err);
+          return reject(err);
+        }
+        resolve(data);
+      });
+    })
+    .then((templateString) => {
+      try {
+        return template(templateString)({
+          manifestEntries: manifestEntries,
+        });
+      } catch (err) {
+        logHelper.error(errors['populating-manifest-tmpl-failed'], err);
+        throw err;
+      }
+    })
+    .then((populatedTemplate) => {
+      return new Promise((resolve, reject) => {
+        fs.writeFile(manifestFilePath, populatedTemplate, (err) => {
+          if (err) {
+            logHelper.error(errors['manifest-file-write-failure'], err);
+            return reject(err);
+          }
+
+          resolve();
+        });
+      });
+    });
+  }
+
+  _filterFiles(files) {
+    // Filter oversize files.
+    files = files.filter((fileDetails) => {
+      if (fileDetails.size > constants.maximumFileSize) {
+        logHelper.warn(`Skipping file '${fileDetails.fil}' due to size. ` +
+          `[Max size supported is ${constants.maximumFileSize}]`);
+        return false;
+      }
+
+      return true;
+    });
+
+    // TODO Filter manifest file itself
+
+    // TODO Filter service worker file itself
+
+    // TODO: Strip prefix
+
+    // TODO: Swap path.sep with '/'
+
+    // Convert to manifest format
+    return files.map((fileDetails) => {
+      return {
+        url: fileDetails.file,
+        revision: fileDetails.hash,
+      };
+    });
+  }
+
+  _getFileManifestDetails(globPattern) {
+    let globbedFiles;
+    try {
+      globbedFiles = glob.sync(globPattern);
+    } catch (err) {
+      logHelper.error(errors['unable-to-glob-files'], err);
+      throw err;
+    }
+
+    const fileDetails = globbedFiles.map((file) => {
+      const fileSize = this._getFileSize(file);
+      if (fileSize === null) {
+        return null;
+      }
+
+      const fileHash = this._getFileHash(file);
+      return {
+        file,
+        hash: fileHash,
+        size: fileSize,
+      };
+    });
+
+    return fileDetails.filter((details) => {
```

Comment:
Another case where collapsing the block into a single-line arrow function would be more concise.